proxy/reverse.go

```diff
@@ -203,3 +209,33 @@ func maybeSetForwardedFor(req *http.Request) {
 	}
 	req.Header.Set("X-Forwarded-For", clientIP)
 }
+
+// checkProxyLoop detects a proxy loop by passing around proxyID
+// and endpoints, and call refresh methods to update endpoints
+// if there's any proxy loop found. It returns true if it has
+// found a proxy loop from a endpoint string.
+func (p *reverseProxy) checkProxyLoop(req *http.Request, ep string) bool {
+	// concatenation of proxyID and endpoint URL string.
+	pv := p.proxyID + "___" + ep
```

Comment:
not really. could you investigate the way that other headers handles this? It could be a good reference for us.